doc/techsupport-manager/core_file_manager.md

```diff
@@ -0,0 +1,329 @@
+
+### Core Manager Requirements
+
+  1. Add a new SONiC service to manage the core dumps as follows: - 
+
+	 a.  Support per-process core file rotation and archiving to optimize disk space 
+
+ 	 b.  Strip Core files sensitive information 
+
+ 2.  Add a new SONIC service to collect and export the tech-support data as follows:-
+
+	 a.  Collect tech-support data when a new core dump is discovered, and export it to a remote server.
+
+	 b. Periodically collect and export tech-support data
+
+
+### Configuration and Management Requirements
+
+To configure the core dump and tech-support data, export to an external server and to view the core details the following config and show commands shall be supported. It is to be noted that the tech-support data always includes the core dumps generated on the system.
+
+### Config commands
+
+>1. Config command to enable/disable the coredump generation of processes.
+>2.  Config command to store the details of exporting tech-support data to an external server which includes remote server name, path, transfer protocol type and the user credentials.
+>2. Config command to enable/disable the tech-support export
+>3. Config command  to configure the tech-support export periodic interval.
+
+### Show commands
+> 1. Show commands to display the core file information
+> 2. show commands to display the tech-support export information.
+ 
+### Scalability Requirements
+There should be a limit on the size of the core file generated and the space occupied on the disk.
+
+### Warmboot Requirements
+> NA
+
+# Functional Description
+
+## Design
+
+The corefile management functionality is divided into two main services. 
+
+	1. Core-dump generation service.
+	2.  Tech-support data export service.
+
+
+## Core-dump generation service
+
+1.  Core files are usually generated when process terminates unexpectedly. Typical conditions are access violations, termination signals (except SIGKILL), etc.,
+2.  ulimit configuration might prevent generation of core due to size configurations. We need to ensure this is not the case.
+3.  Service restart functions - will not generate the core dump as it handle the graceful stop and start. This includes docker service restart as well.
+
+## systemd-coredump
+
+The [systemd-coredump](https://www.freedesktop.org/software/systemd/man/systemd-coredump.html) is a native systemd tool that is available in Debian o/s version 9 (stretch) and above. This tool provides an array of features to manage application core files. When it is installed, as part of base configuration it provides following functionality:
+
+1.  Configures kernel to dump a core when application performs unexpected exit. The process ID, UID, GID, signal received, time of termination, command name of the terminated process are collected. The core dump generate may be affected by ulimit settings. Care should be taken that ulimit settings do not conflict with systemd-coredump configuration.
```

Comment:
Is there a way to add few more data to record maintained per core, like SONiC Version?

Reply:
Can be added